spec/models/miq_expression_spec.rb

```diff
@@ -1631,4 +1631,30 @@
       end
     end
   end
+
+  describe "field_in_sql?" do
+    it "should returns false for model.virtualfield" do
+      field = "ManageIQ::Providers::InfraManager::Vm-archived"
+      expression = {"=" => {"field" => field, "value" => "true"}}
+      expect(described_class.new(expression).field_in_sql?(field)).to eq(false)
```

Comment:
Rats! This stuff is so confusing. So these tests are giving us great design feedback :thumbsup: